File: boj/b_1339.js
function solution(n, str) {
  const cands = Array(26).fill(0);
  let num = 9;
  str.forEach((cand) => {
    for (let i = 0; i < cand.length; i += 1) {
      cands[cand[i].charCodeAt() - 65] += 10 ** (cand.length - i - 1);
    }
  });
  cands.sort((a, b) => b - a);
  for (let i = 0; i < 10; i += 1) {
    if (!cands[i]) break;
    cands[i] *= num--;
  }
  return cands.reduce((accr, curr) => accr + curr, 0);
}
solution(2, ["GCF", "ACDEB"]);

File: boj/b_1339.py
import sys
sys.stdin = open("input.txt", "r") # 제거
input = sys.stdin.readline

n = int(input())
cands, num = [0] * 26, 9
for _ in range(n):
    cand = list(input().rstrip())
    for i in range(len(cand)):
        cands[ord(cand[i]) - 65] += 10 ** (len(cand) - i - 1)
cands.sort(reverse=True)
for i in range(10):
    if not cands[i]:
        break
    cands[i] *= num
    num -= 1
print(sum(cands))